### DISCUSSION TITLE
Origin dropout

### DISCUSSION THREAD

**Discussion by naanlizard** [OvenMediaLabs/OvenMediaEngine#1427] [Help]
We've been debugging edge issues in #1421 but this is new, an origin error where things dropped out for a minute or two

[edge.txt](https://github.com/AirenSoft/OvenMediaEngine/files/13246049/edge.txt)
[origin.txt](https://github.com/AirenSoft/OvenMediaEngine/files/13246050/origin.txt)

Logs centered around the issue

Stream count graph

![image](https://github.com/AirenSoft/OvenMediaEngine/assets/7061721/56acb24e-63f1-4860-833e-05a78237bec2)

now something I find confusing is, at the time of these outbound traffic drops, we see a OUT drop in origin (means we should see a drop in IN traffic in edge), but instead we see the same OUT drop in the edge, and no drop in IN

edge
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/7061721/bf17fa12-af69-43d0-bac7-e041263d5200)

origin
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/7061721/bdb28a87-c10d-4929-8f86-c7924dbad670)

Sorry for the many discussions, but this seemed like a different issue given the origin problem vs edge problems elsewhere

Let me know if you need any more info, I'll post the server.xml and edgeconfig.xml shortly

[Origin Server.xml](https://gist.github.com/naanlizard/7ad3bb701d0168501f8c4c3f06d38fae)

[Edgeconfig.xml](https://gist.github.com/naanlizard/e6112f800456f007e44c38ed8fa6a7a9)

**Comment by getroot** [comment]
I looked at the logs you provided, and it seems that Origin and Edge are installed on the same host. Is that correct? Additionally, are Origin and Edge each running in separate Docker containers?

Based on the symptoms, it appears that this may be related to a socket limit issue. It feels like sockets are being forcibly closed. This could be related to issue #1421 as well.

**Reply by naanlizard** [reply]
Yes they are on the same server. Separate docker containers, and another edge server on another physical server in a different datacenter

**Reply by getroot** [reply]
Do edges on other physical servers draw the same graph?

**Reply by naanlizard** [reply]
Network sockets don't appear to be anywhere near 10k at this time and occasionally spike higher. 

There is a similar drop on the separate edge server, which makes sense as the origin server was not responding

**Comment by getroot** [comment]
From 14:50, the stream count decreased, and the output of the origin and edge decreased accordingly.

The input total for both Edge and Origin maintains around 5mbps. Is the "in" graph normal? Something seems strange.

**Reply by hernanrz** [reply]
That also seems strange to me, the graph is produced by this grafana dashboard (+ some modifications made by us) https://gitlab.com/kukoon/mediathek/ovenmediaengine/helm-chart/-/blob/main/grafana-dashboards/Overview.json?ref_type=heads 

with this exporter

https://github.com/genofire/oven-exporter


this is the prometheus query we are using:

![image](https://github.com/AirenSoft/OvenMediaEngine/assets/13426225/828f57cc-2a06-4e66-a166-409b4228a64c)

**Reply by hernanrz** [reply]
I will review our prometheus query again tomorrow morning in case we are doing something wrong, but it looks correct to me
